napari/_qt/qt_main_window.py

```diff
@@ -531,11 +543,27 @@ def _open_preferences(self):
                 win.resize(self._qt_window._preferences_dialog_size)
 
             self._qt_window._preferences_dialog = win
+            win.valueChanged.connect(self._update_plugin_state)
             win.closed.connect(self._on_preferences_closed)
             win.show()
         else:
             self._qt_window._preferences_dialog.raise_()
 
+    def _update_plugin_state(self):
+        # resetting plugin states in plugin manager
+        for (
+            plugin_name,
+            mod_name,
+            distname,
+        ) in plugins.plugin_manager.iter_available():
```

Comment:
tell me about what's going on here?  `iter_available()` is basically going to look through every package in the environment for the napari `entry_point` (and then it will happen once again in `plugin_manager.discover()` below... 

Are you just trying to unblock all plugins?  If so, you can just call `plugin_manager._blocked.clear()`